week4/assignment/src/components/common/Button.tsx

```diff
@@ -0,0 +1,42 @@
+import styled from '@emotion/styled';
+import { Theme } from '../../styles/theme';
+
+interface ButtonProps {
+  type: 'button' | 'submit'
+  onClick?: () => void;
+  disabled?: boolean;
+  children: React.ReactNode;
+}
+
+const Button = ({ type, onClick, disabled, children }: ButtonProps) => {
+  return (
+    <StyledButton type={type} onClick={onClick} disabled={disabled}>
+      {children}
+    </StyledButton>
+  );
+};
```

Comment:
```suggestion
const Button = ({ children, ...props }: ButtonProps) => {
  return (
    <StyledButton {...props}>
      {children}
    </StyledButton>
  );
};
```

또한 요렇게 props를 통해서 속성을 받아오면, Button 컴포넌트를 더 유연하게 만들고, 코드를 간결하게 작성할 수 있으니 이런 방식도 고려해보면 좋을 것 같습니다!